### PR TITLE
🔨 moved dynamic_sidecar changes from PR1887

### PR DESCRIPTION
**Pull request by GitHK**
## What do these changes do?

Moves changes relative to the `dynamic_sidecar` from https://github.com/ITISFoundation/osparc-simcore/pull/1887

- uses new setting format for `RegistrySettings` added to models_library
- improved health API
- py3.8 upgrade (requirements and minor changes)

### CODE REVIEW

File: services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/core/application.py [hunk 1]
```diff
@@ -1,17 +1,36 @@
...
 from ..models.schemas.application_health import ApplicationHealth
 from .remote_debug import setup as remote_debug_setup
 from .settings import DynamicSidecarSettings
 from .shared_handlers import on_shutdown_handler
+from .utils import login_registry
 
 logger = logging.getLogger(__name__)
 
+#
+# https://patorjk.com/software/taag/#p=display&f=JS%20Stick%20Letters&t=API-server%0A
```
Comment (sanderegg) [inline services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/core/application.py:18]: link points to api-server

**Review by sanderegg — APPROVED** [review]

File: services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/core/application.py (hunk 1 above)
Comment (GitHK) [reply services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/core/application.py]: updated link

File: packages/models-library/src/models_library/settings/docker_registry.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
+from typing import Optional
+
+from pydantic import Field, SecretStr
+
+from .base import BaseCustomSettings
+
```
Comment (pcrespov) [inline packages/models-library/src/models_library/settings/docker_registry.py:6]: now we have settings-library for these things. you did the PR review a few days ago
Actually ``models_library.base.BaseCustomSettings``

File: packages/models-library/src/models_library/settings/docker_registry.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    REGISTRY_USER: str = Field(
+        ..., description="username to access the docker registry"
+    )
+    REGISTRY_PW: SecretStr = Field(
+        ..., description="password to access the docker registry"
+    )
+    REGISTRY_SSL: bool = Field(..., description="access to registry through ssl")
+
+    @property
```
Comment (pcrespov) [inline packages/models-library/src/models_library/settings/docker_registry.py:24]: cached_property (will work when you inherit from the right BaseCustomSettings

File: services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/core/settings.py [hunk 4]
```diff
@@ -2,13 +2,15 @@
 from typing import Any, Optional
 
 from models_library.basic_types import BootModeEnum, PortInt
+from models_library.settings.docker_registry import RegistrySettings
 from pydantic import BaseSettings, Field, PositiveInt, validator
 
 
 class DynamicSidecarSettings(BaseSettings):
```
Comment (pcrespov) [inline services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/core/settings.py:9]: Please read carefully this [PR](https://github.com/ITISFoundation/osparc-simcore/pull/2395) about how the settings work now.

File: packages/models-library/src/models_library/settings/docker_registry.py [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+from .base import BaseCustomSettings
+
+
+class RegistrySettings(BaseCustomSettings):
+
+    REGISTRY_AUTH: bool = Field(..., description="do registry authentication")
+    REGISTRY_PATH: Optional[str] = Field(
+        None, description="development mode only, in case a local registry is used"
+    )
+    REGISTRY_URL: str = Field("", description="url to the docker registry")
```
Comment (pcrespov) [inline packages/models-library/src/models_library/settings/docker_registry.py:14]: AnyUrl vs str?

**Review by pcrespov — CHANGES_REQUESTED** [review]

File: packages/models-library/src/models_library/settings/docker_registry.py (hunk 2 above)
Comment (GitHK) [reply packages/models-library/src/models_library/settings/docker_registry.py:6]: @pcrespov could you link me to that file. And now I am super confused about this one.
There seem sto be package `settings-library` and there is the submodule `models-library.settings` inside `models-library` which are similar. What happened here?

Comment (GitHK) [reply packages/models-library/src/models_library/settings/docker_registry.py:6]: Never mind. I've figured it out. It was placed in the wrong module.

File: packages/models-library/src/models_library/settings/docker_registry.py (hunk 5 above)
Comment (GitHK) [reply packages/models-library/src/models_library/settings/docker_registry.py:14]: unfortunately no, the naming is misleading will add a comment

**Review by pcrespov — APPROVED** [review]

**Pull request merged**